crates/primitives/src/signature/ecdsa_sig.rs

```diff
@@ -0,0 +1,828 @@
+#![allow(unknown_lints, unnameable_types)]
+
+use crate::{
+    hex,
+    signature::{sig::Signature, Parity, SignatureError},
+    uint, U256,
+};
+use alloc::vec::Vec;
+use core::str::FromStr;
+use crate::signature::super_signature::{ArbitrarySuperSig, K256SuperSig, RlpSuperSig, SerdeSuperSig};
+
+/// The order of the secp256k1 curve
+const SECP256K1N_ORDER: U256 =
+    uint!(0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141_U256);
+
+/// An Ethereum ECDSA signature.
+#[derive(Clone, Copy, Debug, Hash, PartialEq, Eq)]
+pub struct EcdsaSignature {
+    v: Parity,
+    r: U256,
+    s: U256,
+}
+
+impl EcdsaSignature {
+    /// Returns the recovery ID.
+    #[cfg(feature = "k256")]
+    #[inline]
+    const fn recid(&self) -> k256::ecdsa::RecoveryId {
+        self.v.recid()
+    }
+
+    #[cfg(feature = "k256")]
+    #[doc(hidden)]
+    #[deprecated(note = "use `Signature::recid` instead")]
+    pub const fn recovery_id(&self) -> k256::ecdsa::RecoveryId {
+        self.recid()
+    }
+
+    #[doc(hidden)]
+    fn test_signature() -> Self {
+        Self::from_scalars_and_parity(
+            b256!("840cfc572845f5786e702984c2a582528cad4b49b2a10b9db1be7fca90058565"),
+            b256!("25e7109ceb98168d95b09b18bbf6b685130e0562f233877d492b94eee0c5b6d1"),
+            false,
+        )
+            .unwrap()
+    }
+}
+
+impl RlpSuperSig for EcdsaSignature {}
+
+impl K256SuperSig for EcdsaSignature {}
+
+impl SerdeSuperSig for EcdsaSignature {}
+
+impl  ArbitrarySuperSig for EcdsaSignature {}
+
+impl Signature for EcdsaSignature {
+    /// Instantiate a new signature from `r`, `s`, and `v` values.
+    fn new(r: U256, s: U256, v: Parity) -> Self {
+        Self { r, s, v }
+    }
+
+    /// Returns the `r` component of this signature.
+    #[inline]
+    fn r(&self) -> U256 {
+        self.r
+    }
+
+    /// Returns the `s` component of this signature.
+    #[inline]
+    fn s(&self) -> U256 {
+        self.s
+    }
+
+    /// Returns the recovery ID as a `u8`.
+    #[inline]
+    fn v(&self) -> Parity {
+        self.v
+    }
+    #[cfg(feature = "rlp")]
+    fn decode_rlp_vrs(buf: &mut &[u8]) -> Result<Self, alloy_rlp::Error> {
+        use alloy_rlp::Decodable;
+
+        let parity: Parity = Decodable::decode(buf)?;
+        let r = Decodable::decode(buf)?;
+        let s = Decodable::decode(buf)?;
+
+        Self::from_rs_and_parity(r, s, parity)
+            .map_err(|_| alloy_rlp::Error::Custom("attempted to decode invalid field element"))
+    }
```

Comment:
on second thought, would be cool to move all the feature gated methods into their respective  XSuperSig trait actually

Reply:
I thought that too!